### PR TITLE
Feat enet cms

### PR DESCRIPTION
**Pull request by apachano**
Authors:
@apachano 

## Summary of changes
-  Add CMS to server
-  Split API between client and server
  
## Caveats
Does this introduce any new bugs?
-  No

## On approval
Please merge

### CODE REVIEW

File: Phoenix/Client/Source/Game.cpp [hunk 1]
```diff
@@ -83,6 +83,10 @@ Game::Game(gfx::Window* window, entt::registry* registry)
 
 	voxels::BlockRegistry::get()->registerAPI(m_modManager);
 
+	m_modManager->registerFunction(
+	    "core.command.register",
+	    [](std::string command, std::string help, sol::function f) {});
```
Comment (SuperFola) [inline Phoenix/Client/Source/Game.cpp:88]: why not use `const std::string&`? same for `sol::function`, would be more cache friendly

File: Phoenix/Server/Source/Server.cpp [hunk 2]
```diff
@@ -42,15 +46,76 @@ Server::Server(std::string save) : m_save(std::move(save))
 	m_game = new Game(&m_registry, &m_running, m_iris);
 }
 
+void registerUnusedAPI(cms::ModManager* manager)
+{
+	manager->registerFunction("core.input.registerInput",
+	                          [](std::string uniqueName,
+	                             std::string displayName,
+	                             std::string defaultKey) {});
```
Comment (SuperFola) [inline Phoenix/Server/Source/Server.cpp:54]: same problem here, `const std::string&` would be more cache friendly

Comment (apachano) [reply Phoenix/Server/Source/Server.cpp:54]: Because constant references to Lua variables starts getting dicey. Last I tried it, it wouldn't work.

Comment (SuperFola) [reply Phoenix/Server/Source/Server.cpp:54]: strange o.o

File: Phoenix/Server/Source/Commander.cpp [hunk 3]
```diff
@@ -40,68 +40,56 @@
...
-	auto m_command = m_commands.find(command);
-	if (m_command != m_commands.end())
+	if (m_commands.find(command) != m_commands.end())
 	{
-		///@TODO log that we over-wrote a command
-		m_command->second.command  = command;
-		m_command->second.help     = help;
-		m_command->second.callback = f;
-		return;
+		std::cout << "Command overwritten: " << command << "\n";
```
Comment (beeperdeeper089) [inline Phoenix/Server/Source/Commander.cpp:92]: Replace with logger.
```
LOG_WARNING("MODDING") << "Command overwritten: " << command;
```

**Pull request merged**